app/scripts/lib/rpc-method-middleware/handlers/add-ethereum-chain.js

```diff
@@ -298,6 +276,27 @@ async function addEthereumChainHandler(
   } catch (error) {
     return end(error);
   }
+  let endpointChainId;
+
+  try {
+    endpointChainId = await jsonRpcRequest(firstValidRPCUrl, 'eth_chainId');
```

Comment:
If either this check or the subsequent one fails, we would want to remove the custom RPC endpoint from the user's wallet because it failed validation.

We should also probably prevent the user from using this network while this validation is ongoing.

Reply:
Update: we talked and decided delaying adding the chain was the easier path. This resolves these concerns.